sklearn/pipeline.py

```diff
@@ -378,9 +392,66 @@ def _check_method_params(self, method, props, **kwargs):
                 fit_params_steps[step]["fit_predict"][param] = pval
             return fit_params_steps
 
+    def _get_step_params(self, *, step_idx, step_params, all_params):
+        """Get params (metadata) for step `name`.
+
+        This transforms the metadata up to this step if required, which is
+        indicated by the `transform_input` parameter.
+
+        If a param in `step_params` is included in the `transform_input` list, it
+        will be transformed.
+
+        `all_params` are the metadata passed by the user. Used to call `transform`
+        on the pipeline itself.
```

Comment:
Adding the Parameters section in the docstring might help to better understand this method.